### PR TITLE
Add tags/labels to GProts

### LINKED ISSUE #11: Tags for GProts
It would be very nice to have labels or tags for GProts, so you can tag them for example as "Probeklausur", "Bonusklausur" or something similar. Hereby, there would be the possibility to filter them by tags and find the GProt you are looking for more easily.

### PR DESCRIPTION
**Pull request by cronala**
This will make it possible to add tags to GProts, for example "Bonusklausur" or "Probeklausur" as described in Issue #11.
Also it will add a column showing whether a GProt contains a pdf file.

### CODE REVIEW

File: mafiasi/gprot/admin.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 from django.contrib import admin
 
-from mafiasi.gprot.models import Attachment, GProt, Notification, Reminder
+from mafiasi.gprot.models import Attachment, GProt, Notification, Reminder, Label
```
Comment (wichmannpas) [inline mafiasi/gprot/admin.py]: This breaks the alphabeticla order of the import items, let your IDE reformat them :)

File: mafiasi/gprot/models.py [hunk 2]
```diff
@@ -6,17 +6,38 @@
...
+class LabelManager(models.Manager):
+
+    def as_choices(self):
+        for label in self.all().order_by('tag'):
+            yield label.pk, label.tag
+
+
+class Label(models.Model):
+    tag = models.CharField(max_length=50)
+    colour = models.CharField(max_length=15, default="default")
```
Comment (wichmannpas) [inline mafiasi/gprot/models.py]: We should check whether we use American or British english in the rest of the code base. I would prefer `color` here, as that is the attribute in CSS etc. as well.

File: mafiasi/gprot/static/img/icon-no.svg [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<svg width="13" height="13" viewBox="0 0 1792 1792" xmlns="http://www.w3.org/2000/svg">
+  <path fill="#dd4646" d="M1277 1122q0-26-19-45l-181-181 181-181q19-19 19-45 0-27-19-46l-90-90q-19-19-46-19-26 0-45 19l-181 181-181-181q-19-19-45-19-27 0-46 19l-90 90q-19 19-19 46 0 26 19 45l181 181-181 181q-19 19-19 45 0 27 19 46l90 90q19 19 46 19 26 0 45-19l181-181 181 181q19 19 45 19 27 0 46-19l90-90q19-19 19-46zm387-226q0 209-103 385.5t-279.5 279.5-385.5 103-385.5-103-279.5-279.5-103-385.5 103-385.5 279.5-279.5 385.5-103 385.5 103 279.5 279.5 103 385.5z"/>
+</svg>
```
Comment (wichmannpas) [inline mafiasi/gprot/static/img/icon-no.svg]: Where do these SVGs come from? Is the license appropriate?

File: mafiasi/gprot/templates/gprot/_basic_form.html [hunk 4]
```diff
@@ -43,3 +43,14 @@
     </div>
     {% endif %}
 </div>
+<div class="formgroup">
+    <label for="labels">{% trans "Tags:" %}</label>
```
Comment (wichmannpas) [inline mafiasi/gprot/templates/gprot/_basic_form.html]: What do we want to call them now, "Label" or "Tag"?

File: mafiasi/gprot/templates/gprot/_basic_form.html [hunk 5]
```diff
@@ -43,3 +43,14 @@
...
+
+    {% for label in form.labels %}
+        <div class="checkbox-inline">
+            {{ label }}
+        </div>
+
```
Comment (wichmannpas) [inline mafiasi/gprot/templates/gprot/_basic_form.html]: Why this empty line?

File: mafiasi/gprot/models.py [hunk 6]
```diff
@@ -6,17 +6,38 @@
...
+
+class LabelManager(models.Manager):
+
+    def as_choices(self):
+        for label in self.all().order_by('tag'):
+            yield label.pk, label.tag
+
+
+class Label(models.Model):
+    tag = models.CharField(max_length=50)
```
Comment (wichmannpas) [inline mafiasi/gprot/models.py]: I would call this field `name`. Or is `tag` not just synonymous to `label` but instead a different semantics?
Also, cf. below regarding whether they are called label or tag – I would prefer using only one of the two terms consistently.

File: mafiasi/gprot/templates/gprot/index.html [hunk 7]
```diff
@@ -40,6 +40,8 @@
         <th>{% trans "Department" %}</th>
         <th>{% trans "Exam date" %}</th>
         <th>{% trans "Examiners" %}</th>
+        <th>{%  trans "Tags" %}</th>
+        <th style="text-align: center">{% trans "PDF?" %}</th>
```
Comment (wichmannpas) [inline mafiasi/gprot/templates/gprot/index.html]: This is a different thing. Move that into another commit.

File: mafiasi/gprot/templates/gprot/index.html [hunk 8]
```diff
@@ -52,6 +54,18 @@
...
         <td>{{ gprot.exam_date|date:"SHORT_DATE_FORMAT" }}</td>
         <td>{{ gprot.examiners|format_examiners }}</td>
+        <td>
+            {% for label in gprot.labels.all %}
+            <span class="label label-{{ label.colour }}">{{ label.tag }}</span>
+            {% endfor %}
+        </td>
+        <td style="text-align: center">
+            {% if gprot.is_pdf %}
+            <img src="{% static "img/icon-yes.svg" %}"/>
```
Comment (wichmannpas) [inline mafiasi/gprot/templates/gprot/index.html]: Does Bootstrap not have such icons already? Is there a need to use image files for it?

File: mafiasi/gprot/templates/gprot/index.html [hunk 9]
```diff
@@ -52,6 +54,18 @@
...
+            {% else %}
+            <img src="{% static "img/icon-no.svg" %}"/>
```
Comment (wichmannpas) [inline mafiasi/gprot/templates/gprot/index.html]: Does Bootstrap not have such icons already? Is there a need to use image files for it?

File: mafiasi/gprot/templates/gprot/list_own.html [hunk 10]
```diff
@@ -31,6 +33,18 @@
         <span class="label label-danger">{% trans "unpublished" %}</span>
         {% endif %}
         </td>
+        <td>
+            {% for label in gprot.labels.all %}
+            <span class="label label-{{ label.colour }}">{{ label.tag }}</span>
+            {% endfor %}
```
Comment (wichmannpas) [inline mafiasi/gprot/templates/gprot/list_own.html]: To prevent the redundancy, this may be moved to a different template file that can be included.

File: mafiasi/gprot/templates/gprot/list_own.html [hunk 11]
```diff
@@ -31,6 +33,18 @@
...
+        </td>
+        <td style="text-align: center">
+            {% if gprot.is_pdf %}
+            <img src="{% static "img/icon-yes.svg" %}"/>
+            {% else %}
+            <img src="{% static "img/icon-no.svg" %}"/>
+            {%  endif %}
```
Comment (wichmannpas) [inline mafiasi/gprot/templates/gprot/list_own.html]: Again, no Bootstrap glyphicon?

File: mafiasi/gprot/admin.py (hunk 1 above)
Comment (cronala) [reply mafiasi/gprot/admin.py]: they are used in the django admin interface

File: mafiasi/gprot/static/img/icon-no.svg (hunk 3 above)
Comment (cronala) [reply mafiasi/gprot/static/img/icon-no.svg]: they are used in the django admin interface

File: mafiasi/gprot/models.py [hunk 12]
```diff
@@ -6,17 +6,38 @@
...
+
 def make_attachment_filename(attachment, filename):
     ext = attachment.mime_type.split('/')[-1]
     return make_filename(attachment, 'gprot-attachment', ext)
 
+
+class LabelManager(models.Manager):
+
+    def as_choices(self):
+        for label in self.all().order_by('name'):
```
Comment (wichmannpas) [inline mafiasi/gprot/models.py]: There is no need for `.all()`, you can call `order_by()` on the manager directly without accessing `all()` first.

Comment (timonegk) [reply mafiasi/gprot/models.py]: For future reference this was copied from https://github.com/fsinfuhh/mafiasi/blob/65381497b80e5b7df9cccde3ecca077c650aa347/mafiasi/teaching/models.py and could be changed there in a separate pull request.

File: mafiasi/gprot/models.py [hunk 13]
```diff
@@ -6,17 +6,38 @@
...
+            yield label.pk, label.name
+
+
+class Label(models.Model):
+    name = models.CharField(max_length=50)
+    color = models.CharField(max_length=15, default="default")
```
Comment (timonegk) [inline mafiasi/gprot/models.py]: I think we should set `choices` to the color field to simplify the selection in the admin interface
